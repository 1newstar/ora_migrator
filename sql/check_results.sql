SET client_min_messages = WARNING;

\dn

\d testschema1.*
\d testschema2.*

\df testschema1.*

SELECT * FROM testschema1.tab1;

SELECT * FROM testschema1.tab2;

SELECT * FROM testschema1.view1;

SELECT * FROM testschema2.tab3;
