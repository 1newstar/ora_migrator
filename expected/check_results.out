SET client_min_messages = WARNING;
\dn
    List of schemas
    Name     |  Owner   
-------------+----------
 public      | postgres
 testschema1 | migrator
 testschema2 | migrator
(3 rows)

\d testschema1.*
                          Table "testschema1.log"
   Column   |            Type             | Collation | Nullable | Default 
------------+-----------------------------+-----------+----------+---------
 username   | character varying(128)      |           | not null | 
 logts      | timestamp(6) with time zone |           | not null | 
 table_name | character varying(128)      |           | not null | 
 id         | numeric                     |           | not null | 
Indexes:
    "log_pkey" PRIMARY KEY, btree (logts, username)

            Index "testschema1.log_pkey"
  Column  |            Type             | Definition 
----------+-----------------------------+------------
 logts    | timestamp(6) with time zone | logts
 username | character varying(128)      | username
primary key, btree, for table "testschema1.log"

                         Sequence "testschema1.seq1"
  Type  | Start | Minimum |       Maximum       | Increment | Cycles? | Cache 
--------+-------+---------+---------------------+-----------+---------+-------
 bigint |     2 |       1 | 9223372036854775807 |         5 | no      |    10

                         Table "testschema1.tab1"
 Column |              Type              | Collation | Nullable | Default 
--------+--------------------------------+-----------+----------+---------
 id     | integer                        |           | not null | 
 vc     | character varying(30)          |           |          | 
 n      | bigint                         |           | not null | 
 bf     | real                           |           |          | 
 bd     | double precision               |           |          | 
 d      | timestamp(0) without time zone |           | not null | 
 ts     | timestamp(6) without time zone |           |          | 
Indexes:
    "tab1_pkey" PRIMARY KEY, btree (id)
    "tab1_bf_bd_idx" btree (bf, bd DESC)
    "tab1_date_part_idx" btree (date_part('day'::text, d))
Check constraints:
    "tab1_bf_check" CHECK (bf > 0::double precision)
Referenced by:
    TABLE "testschema1.tab2" CONSTRAINT "tab2_tab1_id_fkey" FOREIGN KEY (tab1_id) REFERENCES testschema1.tab1(id)
Triggers:
    tab1_trig BEFORE INSERT OR UPDATE ON testschema1.tab1 FOR EACH ROW EXECUTE PROCEDURE testschema1.tab1_trig()

   Index "testschema1.tab1_bf_bd_idx"
 Column |       Type       | Definition 
--------+------------------+------------
 bf     | real             | bf
 bd     | double precision | bd
btree, for table "testschema1.tab1"

          Index "testschema1.tab1_date_part_idx"
  Column   |       Type       |        Definition         
-----------+------------------+---------------------------
 date_part | double precision | date_part('day'::text, d)
btree, for table "testschema1.tab1"

 Index "testschema1.tab1_pkey"
 Column |  Type   | Definition 
--------+---------+------------
 id     | integer | id
primary key, btree, for table "testschema1.tab1"

              Table "testschema1.tab2"
 Column  |  Type   | Collation | Nullable | Default 
---------+---------+-----------+----------+---------
 id      | integer |           | not null | 
 tab1_id | integer |           | not null | 
 c       | text    |           |          | 
 b       | bytea   |           |          | 
Indexes:
    "tab2_pkey" PRIMARY KEY, btree (id)
Foreign-key constraints:
    "tab2_tab1_id_fkey" FOREIGN KEY (tab1_id) REFERENCES testschema1.tab1(id)
Referenced by:
    TABLE "testschema2.tab3" CONSTRAINT "tab3_tab2_id_fkey" FOREIGN KEY (tab2_id) REFERENCES testschema1.tab2(id)

 Index "testschema1.tab2_pkey"
 Column |  Type   | Definition 
--------+---------+------------
 id     | integer | id
primary key, btree, for table "testschema1.tab2"

                    View "testschema1.view1"
 Column |         Type          | Collation | Nullable | Default 
--------+-----------------------+-----------+----------+---------
 vc     | character varying(30) |           |          | 
 c      | text                  |           |          | 

\d testschema2.*
                Table "testschema2.tab3"
 Column  |    Type     | Collation | Nullable | Default 
---------+-------------+-----------+----------+---------
 id      | integer     |           | not null | 
 tab2_id | integer     |           | not null | 
 f       | real        |           |          | 
 ids     | interval(6) |           |          | 
Indexes:
    "tab3_pkey" PRIMARY KEY, btree (id)
Foreign-key constraints:
    "tab3_tab2_id_fkey" FOREIGN KEY (tab2_id) REFERENCES testschema1.tab2(id)

 Index "testschema2.tab3_pkey"
 Column |  Type   | Definition 
--------+---------+------------
 id     | integer | id
primary key, btree, for table "testschema2.tab3"

\df testschema1.*
                             List of functions
   Schema    |   Name    | Result data type | Argument data types |  Type   
-------------+-----------+------------------+---------------------+---------
 testschema1 | tab1_trig | trigger          |                     | trigger
 testschema1 | tomorrow  | date             | d date              | func
(2 rows)

SELECT * FROM testschema1.tab1;
 id |    vc     |   n   |   bf   |   bd   |            d             |            ts            
----+-----------+-------+--------+--------+--------------------------+--------------------------
  1 | some text | 12345 |   3.14 | -2.718 | Fri Jan 26 00:00:00 2018 | Fri Jan 26 22:30:00 2018
  2 |           | 87654 | 9.3452 |        | Fri Dec 29 12:00:00 2017 | 
(2 rows)

SELECT * FROM testschema1.tab2;
 id | tab1_id |      c      |     b      
----+---------+-------------+------------
  1 |       1 | a long text | \xdeadbeef
  2 |       1 |             | \xdeadf00d
(2 rows)

SELECT * FROM testschema1.view1;
    vc     |      c      
-----------+-------------
 some text | a long text
 some text | 
           | 
(3 rows)

SELECT * FROM testschema2.tab3;
 id | tab2_id |  f  |       ids        
----+---------+-----+------------------
  1 |       2 | 2.5 | @ 1 day 12 hours
  2 |       1 |  -1 | @ 1 min 30 secs
(2 rows)

